### DISCUSSION TITLE
Node connections without referencing to the parent container

### DISCUSSION THREAD

**Discussion by kestasli** [terrastruct/d2#1726] [Feature Requests]
Hi, I was testing d2 with some semi-real life cases, it usually takes few iterations to have best grouping of nodes. Currently it is necessary to reference nodes inside container by using container name e.g. ec2 instance in AWS region eu-central-1, zone a1 should be referenced as eu-central-1.a1.server1. This becomes quite a complication if large number of nodes are connected with other nodes and later I decide put some nodes into container. It means that instead of just leaving connections like server1 -- lb1, server2 -- lb2 etc I should redo it to something like eu-central-1.a1.server1 -- lb1 and so on. Don't you think it would be better to allow reference nodes/containers only by name if names are unique? Or maybe there is some workaround already in d2?

**Comment by alixander** [comment]
> Don't you think it would be better to allow reference nodes/containers only by name if names are unique?

No. Scoping is a universal concept in programming and breaking that makes D2 code hard to grok/reason about. What if you have `server` in two containers?

> Or maybe there is some workaround already in d2?

Use globs.

```d2
**.server -> lb1
```

Currently this will create nodes, but with more filters, it won't. Related discussion: https://github.com/terrastruct/d2/issues/1724
